mm2src/trezor/src/client.rs

```diff
@@ -26,55 +26,62 @@ impl TrezorClient {
         T: Transport + Send + Sync + 'static,
     {
         let transport = Box::new(transport);
-        let inner = Arc::new(AsyncMutex::new(TrezorClientImpl {
-            transport,
-            initialized: false,
-        }));
+        let inner = Arc::new(AsyncMutex::new(TrezorClientImpl { transport }));
         TrezorClient { inner }
     }
 
-    /// Initialize a Trezor session by sending
-    /// [Initialize](https://docs.trezor.io/trezor-firmware/common/communication/sessions.html#examples).
-    pub async fn session(&self) -> TrezorResult<TrezorSession<'_>> {
-        {
-            let inner = self.inner.lock().await;
-            if inner.initialized {
-                let mut session = TrezorSession { inner };
-
-                // Check if the device is still alive.
-                session.ping().await?;
-                return Ok(session);
-            }
-        }
-
-        // The device is still not initialized (see [`TrezorClientImpl::initialized`]).
-        // Create new session that calls [`TrezorSession::initialize_device`].
-        self.new_session().await.map(|(_device_info, session)| session)
-    }
-
     /// Initialize a Trezor session by sending
     /// [Initialize](https://docs.trezor.io/trezor-firmware/common/communication/sessions.html#examples).
     /// Returns `TrezorDeviceInfo` and `TrezorSession`.
-    pub async fn new_session(&self) -> TrezorResult<(TrezorDeviceInfo, TrezorSession<'_>)> {
+    pub async fn init_new_session(&self) -> TrezorResult<(TrezorDeviceInfo, TrezorSession<'_>)> {
         let mut session = TrezorSession {
             inner: self.inner.lock().await,
         };
         let features = session.initialize_device().await?;
         Ok((TrezorDeviceInfo::from(features), session))
     }
+
+    /// Checks if the Trezor device is vacant (not occupied),
+    /// and initializes a new Trezor session by sending
+    /// [Initialize](https://docs.trezor.io/trezor-firmware/common/communication/sessions.html#examples).
+    /// Returns `Ok(None)` if it is occupied already.
+    pub async fn try_init_new_session_if_not_occupied(
+        &self,
+    ) -> TrezorResult<Option<(TrezorDeviceInfo, TrezorSession<'_>)>> {
+        let mut session = match self.inner.try_lock() {
+            Some(inner) => TrezorSession { inner },
+            None => return Ok(None),
+        };
+        let features = session.initialize_device().await?;
```

Comment:
Just a question to better understand what is happening, `try_init_new_session_if_not_occupied` is used only in `trezor_connection_status`, why do we have to initialize the device when we are getting the status?

Reply:
There are three ways to check if the device is still connected:
1. Using `webusb` or `libusb` APIs that can provide such info. `webusb` provides an easy way to check the status, but `libusb` requires to sanitize much more actions to use its `hotplug` feature. It's also can be unimplemented on some OS.
2. Trying to ping the device and receive a pong. The problem with this approach is that if the HW device is locked (but still connected), it will require the user to enter a PIN. Even if we cancel the PIN request immediately once the device asked us about it, the device will draw the PIN matrix for a few milliseconds. This looks unfriendly.
3. Trying to initialize a new session by sending `Initialize` packet. It has little overhead since it returns the big [Features](https://github.com/KomodoPlatform/atomicDEX-API/blob/fd5934fd686a92dc2ff46d3047604a54ab012f89/mm2src/trezor/src/proto/messages_management.rs#L21) response. But on the other hand, this request doesn't require the user to enter the PIN, so there is no any interaction on it.

I've chosen the last option as 1) requires more refactoring to implement the `hotplug` feature, and 2) looks unfriendly.
Btw, I just realized that `is_connected` is no more needed, will remove it.
https://github.com/KomodoPlatform/atomicDEX-API/blob/d28c11fb541dae04d1dcfb0200e9172676f309a6/mm2src/crypto/src/hw_ctx.rs#L100-L108

Also I'll add a comment that points to this discussion for better understanding.